src/cpp/session/modules/RStudioAPI.cpp

```diff
@@ -233,6 +234,30 @@ SEXP rs_systemUsername()
    return r::sexp::create(core::system::username(), &protect);
 }
 
+SEXP rs_documentProperties(SEXP idSEXP,
+                           SEXP includeContentsSEXP)
+{
+   // resolve params
+   std::string id = r::sexp::asString(idSEXP);
+   bool includeContents = r::sexp::asLogical(includeContentsSEXP);
+
+   // retrieve document
+   using session::source_database::SourceDocument;
+   boost::shared_ptr<SourceDocument> pDoc(new SourceDocument);
+   Error error = source_database::get(id, pDoc);
+   if (error)
+   {
+      LOG_ERROR(error);
```

Comment:
Should this raise an R error (so that the return value isn't "null" w/o apparent cause from the R caller's point of view?)